### PR TITLE
Changes in Multilayer classes after XPS tests

### PR DESCRIPTION
**Pull request by oridzel**

### CODE REVIEW

File: src/lib/ELF/Drude.m [hunk 1]
```diff
@@ -18,13 +22,18 @@
...
     divisor = mm.^2 + repmat(w_global.^2*gamma^2,1,sq);
 % 	re_oneovereps =1.0+(omega0*omega0)*(w_global*w_global - w_at_q*w_at_q) / divisor;   
     eps_re = mm ./ divisor;
     eps_im = repmat(w_global*gamma,1,sq) ./ divisor;
     
-end
-
-
+    if isIonization
+        eps_re(w_global<w_at_q,:)=0;
```
Comment (Pazus) [inline src/lib/ELF/Drude.m:35]: Look through the Vos article, peak position for Drude function is not there. It’s somwhat like `sqrt(A+wp^2)`

**Review by Pazus — CHANGES_REQUESTED** [review]

**Comment by oridzel** [comment]
Pavel, add this pullrequest

File: src/lib/ELF/eps_sum.m [hunk 2]
```diff
@@ -83,24 +83,28 @@
     for j=1:length(osc.A)
         if isfield(osc,'numion') && j>length(osc.A) - osc.numion
             epsMerm = Mermin(q,w,osc.G(j),osc.Om(j),osc.alpha,true);
+            ind = bsxfun(@ne,epsMerm,0);
```
Comment (Pazus) [inline src/lib/ELF/eps_sum.m]: Здесь не нужен bsxfun, достаточно просто `epsMerm~=0`

File: src/CalculationClasses/MultiLayer/BaseMultiLayer.m [hunk 3]
```diff
@@ -154,6 +154,10 @@ function convLDS (obj,sigma_LDS, alpha_LDS)
     methods %(Access = protected)
         function CalculateEnergyDistribution_E0(obj,theta,phi)
             if nargin < 3; phi = 0; end;
+            if max(phi)>=360 || min(phi)<0
+                error('Elements of phi_mush must be in [0;360) range');
+            end
+            phi = phi+180;
             if ~obj.IsCalculated; error('Not calculated yet.'); end;
```
Comment (Pazus) [inline src/CalculationClasses/MultiLayer/BaseMultiLayer.m:161]: Почему так? Это же повлияет на все ранее сделанные расчёты!

**Review by Pazus — CHANGES_REQUESTED** [review]
Please check throughout the changes for misuses of `bsxfun`. 
If you need element wise power use `omega0.^2`. Remember that all operations between a matrix and a scalar are also elementwise producing result of the matrix size.
The only situation where you need to use `bsxfun` is when you need to perform an operation on column-vector versus row-vector and expect a matrix as the result.

Unfortunately `bsxfun` makes code much harder to read, so we need to reduce its appearance as much as possible.

**Review by Pazus — APPROVED** [review]

**Pull request merged**